lib/teleterm/daemon/daemon.go

```diff
@@ -1074,6 +1076,25 @@ func (s *Service) UpdateUserPreferences(ctx context.Context, clusterURI uri.Reso
 	return preferences, trace.Wrap(err)
 }
 
+// AuthenticateWebDevice is used to upgrade a web session (identified by a DeviceWebToken) to include device trust extensions.
+func (s *Service) AuthenticateWebDevice(ctx context.Context, rootClusterURI uri.ResourceURI, req *api.AuthenticateWebDeviceRequest) (*api.AuthenticateWebDeviceResponse, error) {
+	proxyClient, err := s.GetCachedClient(ctx, rootClusterURI)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+	devicesClient := proxyClient.CurrentCluster().DevicesClient()
+
+	ceremony := dtauthn.NewCeremony()
+	confirmationToken, err := ceremony.RunWeb(ctx, devicesClient, &devicetrustv1.DeviceWebToken{Id: req.DeviceWebToken.Id, Token: req.DeviceWebToken.Token})
```

Comment:
Suggestion: indent fields:

```suggestion
	confirmationToken, err := ceremony.RunWeb(ctx, devicesClient, &devicetrustv1.DeviceWebToken{
		Id: req.DeviceWebToken.Id,
		Token: req.DeviceWebToken.Token,
	})
```